base_url/tests/models_mixin.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Simone Orsi - Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from operator import attrgetter
+
+
+class TestMixin(object):
+    """Mixin to setup fake models for tests.
+
+    Usage - the model:
+
+        class FakeModel(models.Model, TestMixin):
+            _name = 'fake.model'
+
+            name = fields.Char()
+
+    Usage - the test klass:
+
+        @classmethod
+        def setUpClass(cls):
+            super().setUpClass()
+            FakeModel._test_setup_model(cls.env)
+
+        @classmethod
+        def tearDownClass(cls):
+            FakeModel._test_teardown_model(cls.env)
+            super().tearDownClass()
+    """
+
+    # Generate xmlids
+    # This is needed if you want to load data tied to a test model via xid.
+    _test_setup_gen_xid = False
+    # If you extend a real model (ie: res.partner) you must enable this
+    # to not delete the model on tear down.
+    _test_teardown_no_delete = False
+    # You can add custom fields to real models (eg: res.partner).
+    # In this case you must delete them to leave registry and model clean.
+    # This is mandatory for relational fields that link a fake model.
+    _test_purge_fields = []
+
+    @classmethod
+    def _test_setup_model(cls, env):
+        """Initialize it."""
+        cls._build_model(env.registry, env.cr)
+        env.registry.setup_models(env.cr)
+        ctx = dict(env.context, update_custom_fields=True)
+        if cls._test_setup_gen_xid:
+            ctx["module"] = cls._module
+        env.registry.init_models(env.cr, [cls._name], ctx)
+
+    @classmethod
+    def _test_teardown_model(cls, env):
+        """Cleanup registry and real models."""
+
+        for fname in cls._test_purge_fields:
+            model = env[cls._name]
+            if fname in model:
+                model._pop_field(fname)
+
+        if not getattr(cls, "_test_teardown_no_delete", False):
+            del env.registry.models[cls._name]
+            # here we must remove the model from list of children of inherited
+            # models
+            parents = cls._inherit
```

Comment:
@simahawk These lines should interest you. These are required to properly clean all the models when we remove a test model from the registry.....

Reply:
@lmignon ahhh I see. I had a comment on connector search engine in the tests to address this issue (well, to ask you if you had any idea on how to finish the cleanup in such cases :wink: )
Thanks!

Reply:
@lmignon can you propose this fix to conn search engine repo as well? Thanks!

Reply:
@lmignon @simahawk maybe we should move some code in component test to avoid copy/paste

Reply:
I'd like to have it as py package and rely only on py dependency to get it (@lmignon agreed w/ me AFAIR :wink: )
You can live w/ copy/paste for now, meanwhile we decide what to do, where, how.
Maybe an `OCA/test-utils` + auto-publish on pypi?